libraries/chain/fork_database.cpp

```diff
@@ -211,13 +196,13 @@ namespace eosio { namespace chain {
       }
 
       std::ofstream out( fork_db_dat.generic_string().c_str(), std::ios::out | std::ios::binary | std::ofstream::trunc );
-      fc::raw::pack( out, fork_database::magic_number );
-      fc::raw::pack( out, fork_database::max_supported_version ); // write out current version which is always max_supported_version
-      fc::raw::pack( out, *static_cast<block_header_state_legacy*>(&*root) );
+      fc::raw::pack( out, fork_database_t::magic_number );
+      fc::raw::pack( out, fork_database_t::max_supported_version ); // write out current version which is always max_supported_version
+      fc::raw::pack( out, *static_cast<bhs*>(&*root) );             // [greg todo] enought to write only bhs?
```

Comment:
Will need a diff version and need to support old format.

Reply:
Should I make any change here for this PR? 

I'm aware that some work will need to be done to support both formats... probably with a second magic number.

Reply:
No, I think we can merge and fix later.